libcontainer/sync_unix.go

```diff
@@ -14,7 +15,9 @@ import (
 // which ends up making things like json.Decoder hang forever if the packet is
 // bigger than the internal read buffer.
 type syncSocket struct {
-	f *os.File
+	f      *os.File
+	closed bool
+	m      sync.Mutex
```

Comment:
```suggestion
	closed atomic.Bool
```

This might make things a bit nicer...